functions/helpFunctions.tsx

```diff
@@ -191,3 +193,128 @@ export function chooseCorrelatedAssessment(
   }
   return null;
 }
+
+export function uploadDataToLocalstorage() {
+  const totalTasks: number =
+    data.ext_inspera_candidates[0].result.ext_inspera_questions.length;
+  const taskNumbers: number[] = Array.from(Array(totalTasks).keys());
+
+  //check if data is already uploaded to localstorage
+  if (typeof window !== 'undefined') {
+    taskNumbers.map((taskNum: number) => {
+      const key: string = (taskNum + 1).toString() + '_data';
+      const dataToBeUploaded: AnswerType[] = [];
+      // data is not uploaded
+      if (!(key in localStorage)) {
+        data.ext_inspera_candidates.map((candidate: any) => {
+          // create answer objects and add to list
+          dataToBeUploaded.push({
+            assessmentId: uuidv4(),
+            answer: replaceUndefined(
+              candidate.result.ext_inspera_questions[taskNum]
+                .ext_inspera_candidateResponses[0].ext_inspera_response
+            ),
+            candidateId: parseInt(candidate.result.ext_inspera_candidateId),
+            maxPoints:
+              candidate.result.ext_inspera_questions[taskNum]
+                .ext_inspera_maxQuestionScore,
+            taskNumber: taskNum + 1,
+          });
+        });
+
+        //add to localstorage
+        localStorage.setItem(key, JSON.stringify(dataToBeUploaded));
+      }
+    });
+  }
+}
+
+export function getApprovedAssessments(taskNum: string): ApprovalType[] {
+  const approvedKey: string = taskNum + '_approved';
+  let approvedAssessments: ApprovalType[] = [];
+  if (typeof window !== 'undefined') {
+    if (approvedKey in localStorage) {
+      approvedAssessments = JSON.parse(
+        localStorage.getItem(approvedKey) as string
+      );
+    }
+  }
+  return approvedAssessments;
+}
+
+export function getRawAssessments(taskNum: string): AnswerType[] {
+  const rawKey: string = taskNum + '_data';
+  let rawAssessments: AnswerType[] = [];
+  if (typeof window !== 'undefined') {
+    if (rawKey in localStorage) {
+      rawAssessments = JSON.parse(localStorage.getItem(rawKey) as string);
+    }
+  }
+  return rawAssessments;
+}
+
+export function getStartedAssessments(taskNum: string): AssessmentType[] {
+  const startedKey: string = taskNum + '_assessments';
+  let startedAssessments: AssessmentType[] = [];
+  if (typeof window !== 'undefined') {
+    if (startedKey in localStorage) {
+      startedAssessments = JSON.parse(
+        localStorage.getItem(startedKey) as string
+      );
+    }
+  }
+  return startedAssessments;
+}
+
+export function getAssessmentData(taskNum: string) {
```

Comment:
I personally think that you should get all answers regardless of if you have started the assessment or not (ref. `else {} `). Especially since you get approved assessments even though that something the user shouldn't get on assessment page (due to rerouting).

Reply:
Ooor, if we dont have the back button on assessment page, I think its nice like this:)

Reply:
> I personally think that you should get all answers regardless of if you have started the assessment or not (ref. `else {} `). Especially since you get approved assessments even though that something the user shouldn't get on assessment page (due to rerouting).

Not sure if I understand this right, but if you get all answers  you would not continue where you left off, but start from the start again?

Reply:
Not sure if I understand you haha, but I just meant that you get all assessments, and those that were assessed gets the correct score and flag. And when the user goes back to assessment it jumps to current page in the same way that it happens when changing maxItemsPerPage

Reply:
Ahhaa, I think I understand it now 💡